docker/Dockerfile

```diff
@@ -0,0 +1,52 @@
+# Docker image with a build toolchain and environment variables set to use
+# the wasi-sdk sysroot. The SDK distribution must have first been built,
+# for example using docker_build.sh
+
+# Extract built SDK archive to copy out the sysroot
+FROM ubuntu:22.04 as dist
+
+ADD dist/wasi-sdk-*.*-linux.tar.gz /
+
+# Move versioned folder to unversioned to using bash glob to allow
+# this file to be independent of major version number.
+RUN mv /wasi-sdk-* /wasi-sdk
+
+# Use ubuntu to use official repository with newer cmake packages
+FROM ubuntu:22.04
```

Comment:
can you explain the two different "FROM ubuntu" lines above?

Reply:
Tried extending the comment above the first one, it's a multi-stage dockerfile, we don't need the entire SDK so we extract in a prepatory stage to reduce docker image size

Reply:
nice.